### ISSUE TITLE
reactCookie.remove not functioning.

### ISSUE THREAD

**Issue by jcdavison** [eXon/react-cookie#16]
I can't believe this is actually a problem with the library, but is there some condition in my application that could be causing behavior like what I'm observing in this console screenshot? 

https://www.dropbox.com/s/805pyfznabpyr54/Screenshot%202016-02-02%2012.57.18.png?dl=0

Other pages in the application appear to be working as expected.

**Comment by eXon** [comment]
If you do a page refresh, is the cookie removed?

**Comment by jcdavison** [comment]
no

**Comment by jcdavison** [comment]
The one detail that I notice is that the cookie will clear properly with non dynamic route

`/labs` then in helper module `Cookie.remove(token)` works 
bu
`/labs/:id` `Cookie.remove(token)` does not work.

Any thought on what is causing this?

For context, I've got a react app using relay in conjunction with a rails backend.

**Comment by eXon** [comment]
This is probably a path problem with your cookie. Make sure to do:

```javascript
cookie.save('token', value, { path: '/' }); 
// ...
cookie.remove('token', { path: '/' });
```

Browsers are bitches about that.

**Comment by sushilkumar48** [comment]
Same occurred to me a day ago. Passing { path: '/' } solves the problem.

Thanks a lot!

**Comment by zachfejes** [comment]
@eXon that was SUPER helpful! Thanks :)

**Comment by destinatis** [comment]
Does not work. >:(

**Comment by eXon** [comment]
@destinatis can you be more precise about your issue? Please open a new one with steps to reproduce.

**Comment by yaskovdev** [comment]
To me it started removing only after I specified _both_ path and domain:
```javascript
cookies.remove('token', { domain: COOKIE_DOMAIN, path: COOKIE_PATH })
```

**Comment by kimuyjoong** [comment]
> To me it started removing only after I specified _both_ path and domain:
> 
> ```js
> cookies.remove('token', { domain: COOKIE_DOMAIN, path: COOKIE_PATH })
> ```

Super help for me! thanks !

**Comment by besil** [comment]
I'm experiencing the issue. Didn't help adding domain or path

`new Cookies().remove('token', {path: "/", domain: "localhost"});`

I'm using Firefox Developer Edition in private mode.

With chrome, it's working fine, even without the domain option.

I suppose @eXon's comment "Browsers are bitches about that" is more true than I believed

**Comment by markjackson02** [comment]
I'm having a similar problem except I can't even get it to remove in Chrome... @besil this issue is closed so you might want to open a new issue.

**Comment by MatthieuStadelmann** [comment]
I'm having a similar problem too. Did you find a solution @markjackson02 @besil ?

**Comment by besil** [comment]
Found out FF extensions did cause interference.
After disabiling all of them, cookie deletion started working.

In Chrome everything worked fine (I have no extensions there)

**Comment by MatthieuStadelmann** [comment]
Thanks for answering. Well, I think I have the same problem on Chrome where I have extensions but on Firefox it works fine.

**Comment by markjackson02** [comment]
@MatthieuStadelmann My workaround is to use Express's `res.clearCookie()` which clears the client cookie when it receives the response

**Comment by MatthieuStadelmann** [comment]
Thanks @markjackson02 

However, I don't use express in my React application. I need to be able to remove the cookies when the user logs out.

**Comment by markjackson02** [comment]
@MatthieuStadelmann Can you use LocalStorage instead of cookies? In any case, I think this is deserving of a new issue. This one is closed so probably none of the devs of this package will see this thread.

**Comment by MatthieuStadelmann** [comment]
@markjackson02 No, for security purpose I can't. But you're right, I'll open a new thread if I can't find a solution :-)

**Comment by kennethbrent** [comment]
I had the same issue and could not figure out the problem - setting domain and path was not working. I don't know why but once i changed my function to an arrow function it works as expected.
`
signOut(){
 Cookies.remove('cookies_name')
}
to
signOut = () => {
 Cookies.remove('cookies_name')
}
`

**Comment by Boby** [comment]
for me I used <code>removeCookie('yourcookiename', { domain: '', path: '/' });</code> and it clears out but I had to refresh, anyway we can get the cookie domain?

**Comment by inthelamp** [comment]
RemoveCookie function removed the cookie, but I had to refresh a signout component which called the function and solve an infinite loop problem out of the component because of the refreshing.

**Comment by yvzkr** [comment]
> This is probably a path problem with your cookie. Make sure to do:
> 
> ```js
> cookie.save('token', value, { path: '/' }); 
> // ...
> cookie.remove('token', { path: '/' });
> ```
> 
> Browsers are bitches about that.

you are king

**Comment by 6thpath** [comment]
> This is probably a path problem with your cookie. Make sure to do:
> 
> ```js
> cookie.save('token', value, { path: '/' }); 
> // ...
> cookie.remove('token', { path: '/' });
> ```
> 
> Browsers are bitches about that.

This works, ty

**Comment by RutvikStackDev** [comment]
> This is probably a path problem with your cookie. Make sure to do:
> 
> ```js
> cookie.save('token', value, { path: '/' }); 
> // ...
> cookie.remove('token', { path: '/' });
> ```
> 
> Browsers are bitches about that.

This solution has a flaw. After logged out if you refresh the page then it will login you back. At-least happened to me several times.

Please others also check if this is a case or not.

**Comment by egalindoEH** [comment]
This happened to me in latest version because I deleted setCookie in const [cookie, removeCookie] = useCookies([...]);

So removeCookie was being used as setCookie, just add it again: const [cookie, **setCookie**, removeCookie] = useCookies([...]);

**Comment by raibove** [comment]
> This happened to me in latest version because I deleted setCookie in const [cookie, removeCookie] = useCookies([...]);
> 
> So removeCookie was being used as setCookie, just add it again: const [cookie, **setCookie**, removeCookie] = useCookies([...]);

Thank you, this worked for me

**Comment by Nilinofficial** [comment]
I did the same but the cookie is removing only after clicking it twice on the first click it removes and adds cookie back automatically even if i add setCookie
